### ISSUE TITLE
Apply Context on HUD Modal

### ISSUE THREAD

**Issue by rocknrolaf** [vaersaagod/matrixmate#10]
Hi!
First of all really nice plugin! There is one issue when using the modal HUD on entries fields and creating new entries which have Matrix Fields via the button in the modal.

I tried adding a piece of code in the init method which seems to work:

```javascript
// Change context when the modal HUD is used to create a new entry
Garnish.on(
    Craft.BaseElementEditor,
    "showHud",
    $.proxy(function(e) {
        this.settings.context = 'entryType:' + e.target.settings.attributes.typeId;
    }, this)
);
```
Maybe you could take a look. Thank You!

**Comment by mmikkel** [comment]
Hi @rocknrolaf, thanks a lot for contributing!

Yes, there's definitely issues with element editor modals. Thanks a lot for the code snippet – your solution seems to work when creating new entries, but not for editing existing ones. Also, we should probably support category editor modals, as well.

Hopefully I'll be able to figure a fix shortly.

**Comment by mmikkel** [comment]
Hi @rocknrolaf – this issue should be resolved in [MatrixMate 1.2.2](https://github.com/vaersaagod/matrixmate/releases/tag/v1.2.2).
